lib/ansible/modules/storage/glusterfs/gluster_peer.py

```diff
@@ -0,0 +1,173 @@
+    force:
+       type: bool
+       default: "false"
+       description:
+          - Applicable only while removing the nodes from the pool. gluster
+            will refuse to detach a node from the pool if any one of the node
+            is down, in such cases force can be used.
+requirements:
+  - GlusterFS > 3.2
+notes:
+  - This module does not support check mode.
+'''
+
+EXAMPLES = '''
+- name: Create a trusted storage pool
+  gluster_peer:
+        state: present
+        nodes:
+             - 10.0.1.5
+             - 10.0.1.10
+
+- name: Delete a node from the trusted storage pool
+  gluster_peer:
+         state: absent
+         nodes:
+              - 10.0.1.10
+
+- name: Delete a node from the trusted storage pool by force
+  gluster_peer:
+         state: absent
+         nodes:
+              - 10.0.0.1
+         force: true
+'''
+
+RETURN = '''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from distutils.version import LooseVersion
+
+
+class Peer(object):
+    def __init__(self, module):
+        self.module = module
+        self.state = self.module.params['state']
+        self.nodes = self.module.params['nodes']
+        self.action = ''
+        self.force = ''
+
+    def gluster_peer_ops(self):
+        if not self.nodes:
+            self.module.fail_json(msg="nodes list cannot be empty")
+        self.force = 'force' if self.module.params.get('force') else ''
+        if self.state == 'present':
+            self.nodes = self.get_to_be_probed_hosts(self.nodes)
+            self.action = 'probe'
+            # In case of peer probe, we do not need `force'
+            self.force = ''
+        else:
+            self.action = 'detach'
+        self.call_peer_commands()
+
+    def get_to_be_probed_hosts(self, hosts):
+        rc, output, err = self.module.run_command("gluster pool list")
+        peers_in_cluster = [line.split('\t')[1].strip() for
+                            line in filter(None, output.split('\n')[1:])]
+        try:
+            peers_in_cluster.remove('localhost')
+        except ValueError:
+            # It is ok not to have localhost in list
+            pass
+        hosts_to_be_probed = [host for host in hosts if host not in
+                              peers_in_cluster]
+        return hosts_to_be_probed
+
+    def call_peer_commands(self):
+        result = {}
+        result['msg'] = ''
+
+        for node in self.nodes:
+            rc, out, err = self._run_command('gluster', ' peer %s %s %s'
+                                             % (self.action, node, self.force))
+            if rc:
+                result['rc'] = rc
+                result['msg'] = err
+                # Fail early, do not wait for the loop to finish
+                self.module.fail_json(**result)
+            else:
+                result['rc'] = rc
+                if 'already in peer' in out or \
+                   'localhost not needed' in out:
+                    result['changed'] = False
```

Comment:
`result['changed']` must be set to `False` before the loop, then result from previous iteration must not be lost, use something like `result['changed'] |= False`

Reply:
You're right, thanks. That was a good one.